crates/oxc_semantic/src/builder.rs

```diff
@@ -2016,9 +2013,7 @@ impl<'a> SemanticBuilder<'a> {
                 self.class_table_builder.pop_class();
             }
             AstKind::ExportSpecifier(_) => {
-                if !self.current_reference_flags.is_type_only() {
-                    self.current_reference_flags = ReferenceFlags::empty();
-                }
+                self.current_reference_flags = ReferenceFlags::empty();
```

Comment:
Should I use

```
s.export_kind.is_type()
                    || matches!(self.nodes.parent_kind(self.current_node_id), Some(AstKind::ExportNamedDeclaration(decl)) if decl.export_kind.is_type())
```

and only `self.current_reference_flag-=ReferenceFlags::Type`.

Reply:
We may need to handle this in `visit_export_specifier`; I think checking the parental `ExportNamedDeclaration` every time here is inefficient. 

You can refer https://github.com/oxc-project/oxc/blob/877a83a81bdf93dcb890df174132cfff381a58d1/crates/oxc_semantic/src/builder.rs#L1777-L1812 to do that

Reply:
Yeah. I thought of this too. But it seems can't be done to check `if previous item is a type` to determine if this item is a type. You have to check it's parent.

Maybe we could store the check result of ` parental ExportNamedDeclaration`? But that would lead to add a extra field. To reduce changes, I didn't do it in the first place.